api/star-wars-films/src/App.js

```diff
@@ -0,0 +1,114 @@
+import React, { Component } from 'react';
+import './App.css';
+import { Link } from 'react-router-dom';
+
+// export const About = () => <p>Explore Star Wars Characters</p>;
+
+
+export default class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      results: {},
+      page: 1,
+      people: [],
+      shown: true
+    };
+  }
+
+  
+  componentWillMount() {
+    this.getCharacters();
+  }
+
+  getCharacters = () => {
+    fetch(`https://swapi.co/api/people/?page=${this.state.page}`)
+      .then(response => response.json())
+      .then(data => {
+        this.setState(() => {
+          return {
+            results: data
+          };
+        });
+      });
+  }
+
+  toggleHidden() {
+    this.setState({
+      isHidden: !this.state.isHidden
+    })
+  }
+
+  next = () => {
+    this.setState((prevState) => {
+      return { page: prevState.page + 1 };
+    }, this.getCharacters);
+  }
+
+  previous = () => {
+    this.setState((prevState) => {
+      return { page: prevState.page - 1, shown: false };
+    }, this.getCharacters);
+  }
+
+
+
+  render() {
+
+      if (this.state.page < 2) {
```

Comment:
if.. else if.. else blocks have a whole lot in common.   The only difference seem to be buttons at the bottom.   Maybe eliminate repetition by wrapping if..else if..else only around the buttons